internal/sql/repository/bulkUpdate/BulkUpdateRepository.go

```diff
@@ -0,0 +1,161 @@
+	for i, appNameExclude := range appNameExcludes {
+		if i == 0 {
+			appNameExcludesQuery += fmt.Sprintf("app_name NOT LIKE '%s' ", appNameExclude)
+		} else {
+			appNameExcludesQuery += fmt.Sprintf("AND app_name NOT LIKE '%s' ", appNameExclude)
+		}
+	}
+	appNameQuery := fmt.Sprintf("( %s ) AND ( %s )", appNameIncludesQuery, appNameExcludesQuery)
+	return appNameQuery
+}
+
+func (repositoryImpl BulkUpdateRepositoryImpl) FindBulkUpdateReadme(operation string) (*BulkUpdateReadme, error) {
+	bulkUpdateReadme := &BulkUpdateReadme{}
+	err := repositoryImpl.dbConnection.
+		Model(bulkUpdateReadme).Where("operation LIKE ?", operation).
+		Select()
+	return bulkUpdateReadme, err
+}
+
+func (repositoryImpl BulkUpdateRepositoryImpl) FindBulkAppNameForGlobal(appNameIncludes []string, appNameExcludes []string) ([]*pipelineConfig.App, error) {
+	apps := []*pipelineConfig.App{}
+	if len(appNameIncludes) == 0 || len(appNameExcludes) == 0 {
+		return apps, nil
+	}
+	appNameQuery := repositoryImpl.BuildAppNameQuery(appNameIncludes, appNameExcludes)
+	err := repositoryImpl.dbConnection.
+		Model(&apps).Join("INNER JOIN charts ch ON app.id = ch.app_id").
+		Where(appNameQuery).
+		Where("app.active = ?", true).
+		Where("ch.latest = ?", true).
+		Select()
+	return apps, err
+}
+
+func (repositoryImpl BulkUpdateRepositoryImpl) FindBulkAppNameForEnv(appNameIncludes []string, appNameExcludes []string, envId int) ([]*pipelineConfig.App, error) {
+	apps := []*pipelineConfig.App{}
+	if len(appNameIncludes) == 0 || len(appNameExcludes) == 0 {
+		return apps, nil
+	}
+	appNameQuery := repositoryImpl.BuildAppNameQuery(appNameIncludes, appNameExcludes)
+	err := repositoryImpl.dbConnection.
+		Model(&apps).Join("INNER JOIN charts ch ON app.id = ch.app_id").
+		Join("INNER JOIN chart_env_config_override ON ch.id = chart_env_config_override.chart_id").
+		Where(appNameQuery).
+		Where("app.active = ?", true).
+		Where("chart_env_config_override.target_environment = ? ", envId).
+		Where("chart_env_config_override.latest = ?", true).
+		Select()
+	return apps, err
+}
+func (repositoryImpl BulkUpdateRepositoryImpl) FindBulkChartsByAppNameSubstring(appNameIncludes []string, appNameExcludes []string) ([]*chartConfig.Chart, error) {
+	charts := []*chartConfig.Chart{}
+	if len(appNameIncludes) == 0 || len(appNameExcludes) == 0 {
+		return charts, nil
+	}
+	appNameQuery := repositoryImpl.BuildAppNameQuery(appNameIncludes, appNameExcludes)
+	err := repositoryImpl.dbConnection.
+		Model(&charts).Join("INNER JOIN app ON app.id=app_id ").
+		Where(appNameQuery).
+		Where("app.active = ?", true).
+		Where("latest = ?", true).
+		Select()
+	return charts, err
+}
+func (repositoryImpl BulkUpdateRepositoryImpl) FindBulkChartsEnvByAppNameSubstring(appNameIncludes []string, appNameExcludes []string, envId int) ([]*chartConfig.EnvConfigOverride, error) {
+	charts := []*chartConfig.EnvConfigOverride{}
+	if len(appNameIncludes) == 0 || len(appNameExcludes) == 0 {
```

Comment:
len(appNameExcludes) == 0 is acceptable

Reply:
also this is business logic validation should be outside  repository

Reply:
handled by @vikramdevtron